controllers/selfnoderemediation_controller.go

```diff
@@ -212,27 +212,40 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 	return result, r.updateSnrStatusLastError(snr, err)
 }
 
-func (r *SelfNodeRemediationReconciler) updateSnrProcessingCondition(reason processingChangeReason, snr *v1alpha1.SelfNodeRemediation) error {
-	var conditionStatus metav1.ConditionStatus
-	switch reason {
+func (r *SelfNodeRemediationReconciler) updateSnrTypeConditions(processingTypeReason processingChangeReason, snr *v1alpha1.SelfNodeRemediation) error {
+	var processingConditionStatus, succeededConditionStatus metav1.ConditionStatus
+	switch processingTypeReason {
 	case remediationStarted:
-		conditionStatus = metav1.ConditionTrue
-	case remediationFinished, remediationTerminatedByNHC:
-		conditionStatus = metav1.ConditionFalse
+		processingConditionStatus = metav1.ConditionTrue
+		succeededConditionStatus = metav1.ConditionUnknown
+	case remediationFinished:
+		processingConditionStatus = metav1.ConditionFalse
+		succeededConditionStatus = metav1.ConditionTrue
+	case remediationTerminatedByNHC:
+		processingConditionStatus = metav1.ConditionFalse
+		succeededConditionStatus = metav1.ConditionFalse
 	default:
-		err := fmt.Errorf("unkown processingChangeReason:%s", reason)
+		err := fmt.Errorf("unkown processingChangeReason:%s", processingTypeReason)
 		r.Log.Error(err, "couldn't update snr processing condition")
 		return err
 	}
 
-	if meta.IsStatusConditionPresentAndEqual(snr.Status.Conditions, v1alpha1.SnrConditionProcessing, conditionStatus) {
+	if meta.IsStatusConditionPresentAndEqual(snr.Status.Conditions, v1alpha1.ProcessingTypeCondition, processingConditionStatus) {
```

Comment:
what about the Succeeded condition?

Reply:
Maybe I'm missing something but I did think about it. 
Originally the idea was to skip update if Processing condition was not change, logically IIUC there is no such use case where Success condition would change and Processing condition will not so current condition still good enough.

Reply:
remediationFinished and remediationTerminatedByNHC hace the same Processing but different Succeeeded status?

Reply:
They do, but this is not my point.
The point that I'm arguing is that Succedded condition will change only when Processing status also changes.

Reply:
>  Succeeded will change only when Processing also changes

when we first have remediationFinished and then remediationTerminatedByNHC or the other way around, Processing will stay at False, but Succeeded changes. Even when we know that this transition is unlikely or even impossible to happen today, that might change. Or we can get new reasons, resulting in other status combinations. IMHO this is is a maintenance burden and/or a potential future bug.

Reply:
IMO both are fine, on one hand I don't see an issue with the current code but on the other hand the code change is very simple and risk free so I'll be happy to oblige :smile: